File: src/python4cpm/secrets.py
from python4cpm.crypto import Crypto


class Secret:
    def __init__(self, secret: str) -> None:
        self._secret = secret

    def get(self) -> str:
        if Crypto.ENABLED and self._secret:
            return Crypto.decrypt(self._secret)
        else:
            return self._secret


class Secrets:
    SECRETS = (
        "password",
        "logon_password",
        "reconcile_password",
        "new_password"
    )

    def __init__(
        self: str,
        password: str,
        logon_password: str,
        reconcile_password: str,
        new_password: str
    ) -> None:
        self._password = Secret(password)
        self._logon_password = Secret(logon_password)
        self._reconcile_password = Secret(reconcile_password)
        self._new_password = Secret(new_password)

    @property
    def password(self) -> str:
        return self._password

    @property
    def new_password(self) -> str:
        return self._new_password

    @property
    def logon_password(self) -> str:
        return self._logon_password

    @property
    def reconcile_password(self) -> str:
        return self._reconcile_password

File: src/plugin/Secrets.cs
using System.Security;

namespace CyberArk.Extensions.Plugin.Python4CPM
{
    public class Secret
    {
        public Secret(string value)
        {
            Value = value;
        }

        public string Value { get; }

        public override string ToString()
        {
            if (Value != null)
                return "[ENCRYPTED]";
            return "[NOT SET]";
        }
    }

    public class Password : Secret
    {
        public Password(string value) : base(value) { }

        public static Password FromSecureString(SecureString secureStr)
        {
            Secret secret = Crypto.Encrypt(secureStr);
            return new Password(secret.Value);
        }
    }

    public class NewPassword : Secret
    {
        public NewPassword(string value) : base(value) { }

        public static NewPassword FromSecureString(SecureString secureStr)
        {
            Secret secret = Crypto.Encrypt(secureStr);
            return new NewPassword(secret.Value);
        }
    }
}
